cmd/influxd/launcher/storage_test.go

```diff
@@ -152,3 +155,18 @@ func TestLauncher_BucketDelete(t *testing.T) {
 		t.Fatalf("after bucket delete got %d, exp %d", got, exp)
 	}
 }
+
+func TestLauncher_UpdateRetentionPolicy(t *testing.T) {
+	l := launcher.RunTestLauncherOrFail(t, ctx, nil)
+	l.SetupOrFail(t)
+	defer l.ShutdownOrFail(t, ctx)
+
+	bucket, err := l.BucketService(t).FindBucket(ctx, influxdb.BucketFilter{ID: &l.Bucket.ID})
+	require.NoError(t, err)
+	require.NotNil(t, bucket)
+
+	newRetentionPeriod := 1 * time.Hour
+	bucket, err = l.BucketService(t).UpdateBucket(ctx, bucket.ID, influxdb.BucketUpdate{RetentionPeriod: &newRetentionPeriod})
+	require.NoError(t, err)
+	assert.Equal(t, bucket.RetentionPeriod, newRetentionPeriod)
+}
```

Comment:
I see this verifying the `RetentionPeriod` and there doesn't seem to be a way to verify that the `ShardGroupDuration` behaved as you expected. It might be worth setting another expected value here (other than 1 hour), because I see this codepath: 

https://github.com/influxdata/influxdb/blob/025319c3875a73ede0345687e256ebc3249318ab/v1/services/meta/data.go#L1279-L1281

Which returns `1 * time.Hour` as a default here:

https://github.com/influxdata/influxdb/blob/025319c3875a73ede0345687e256ebc3249318ab/v1/services/meta/data.go#L1266-L1274

Reply:
I thought about that, but we're not really testing the `meta.Client` as we're a consumer of that API in this code. I verified that `UpdateRetentionPolicy` API is tested here:

https://github.com/influxdata/influxdb/blob/025319c3875a73ede0345687e256ebc3249318ab/v1/services/meta/client_test.go#L463-L522

The intent of our test is to confirm we're able to set the lowest value and that the `UpdateRetentionPolicy` doesn't complain with an error. I do think will want to adjust the rules for `ShardGroupDuration` and / or allow the users to modify it themselves, just as they can in 1.x. Probably via CLI tooling and 2.0 OSS specific API